bucket_map/src/bucket_storage.rs

```diff
@@ -121,13 +90,7 @@ impl BucketStorage {
             capacity_pow2,
             stats,
             max_search,
-        }
-    }
-
-    /// non-zero if there is a header allocated prior to each element to store the 'allocated' bit
-    fn header_size() -> usize {
-        match IS_ALLOCATED_FLAG_LOCATION {
-            IsAllocatedFlagLocation::InHeader => std::mem::size_of::<Header>(),
+            contents: O::new(1 << capacity_pow2),
```

Comment:
current impl using header does not need to allocate anything that needs to know the capacity of elements. Other implementations, such as bitvec need to know capacity of elements. instead of using ::default() here and then changing it later, it seems cleaner to define the trait to use fn new and then just implement default behavior for this particular mechanism of implementing occupied/free.